app/src/lib/repository-matching.ts

```diff
@@ -148,3 +148,29 @@ export function urlMatchesRemote(url: string | null, remote: IRemote): boolean {
     caseInsensitiveEquals(remoteUrl.name, cloneUrl.name)
   )
 }
+
+/**
+ * Match a URL-like string to the Clone URL of a GitHub Repository
+ *
+ * @param url A remote-like URL to verify against the existing information
+ * @param gitHubRepository GitHub API details for a repository
+ */
+export function urlMatchesCloneURL(
```

Comment:
I settled on this function because it ended up feeling very similar to `urlMatchesRemote` which is also defined here, but compares the `url` string to an `IRemote`.